src/relay/backend/contrib/codegen_c/codegen.cc

```diff
@@ -40,35 +40,44 @@ using namespace backend;
  * purpose. Only several binary options are covered. Users
  * may need to extend them to cover more operators.
  */
-class CodegenC : public ExprVisitor, public CodegenCBase {
+class CodegenC : public relay::ExprFunctor<std::vector<Output>(const Expr&)>,
```

Comment:
I'm debating if we really need to return a vector instead of a single output, because we always return a single value even for tuple nodes.

Reply:
I think its' better to keep vector. We flatten tuple and may need to have multiple return values. No need to refactor in that case.